device_attributes/index.bs

```diff
@@ -0,0 +1,192 @@
+Repository: https://github.com/WICG/WebApiDevice
+Group: wicg
+</pre>
+ 
+<pre class=link-defaults>
+spec:html; type:interface; text:Navigator
+spec:url; type:dfn; text:origin
+spec:html; type:dfn; for:/; text:global object
+spec:webidl; type:dfn; text:resolve
+spec:webidl; type:dfn; text:record
+</pre>
+ 
+<pre class="anchors">
+ 
+</pre>
+ 
+# Model # {#model}
+ 
+## Managed devices ## {#managed-devices}
+ 
+The API is presumed to be used on devices which are not fully controlled by the end user, but rather by an external entity, the device administrator. Device administrators are given the power to fully control the managed devices.
+ 
+## Managed web applications ## {#managed-web-applications}
+ 
+The API is presumed to be used by web applications, which are understood and managed by the device administrators.
+ 
+## Device attributes ## {#device-attributes}
+ 
+The API is presumed to be used with a management infrastructure which allows administrators to manage a fleet of devices and configure properties for those devices.
+ 
+### <dfn>Annotated Asset ID</dfn> ### {#annotated-asset-id}
+
+An administrator-defined value which uniquely identifies a device within an organization.
+
+### <dfn>Annotated Location</dfn> ### {#annotated-location}
+
+An administrator-defined value which uniquely identifies a location within an organization.
+
+### <dfn>Directory ID</dfn> ### {#directory-id}
+ 
+An inventory management system-defined value which uniquely identifies a device within an organization.
+ 
+### <dfn>Hostname</dfn> ### {#hostname}
+ 
+An administrator-defined value which is used as the device hostname during DHCP requests.
+ 
+### <dfn>Serial Number</dfn> ### {#serial-number}
+ 
+A manufacturer-defined value which uniquely identifies a device among those produced by that manufacturer.
+ 
+# {{NavigatorManagedData}} interface # {#navigatormanageddata-interface}
+ 
+<xmp class="idl">
+[
+  SecureContext,
+  Exposed=Window
+] interface NavigatorManagedData : EventTarget {
+  // Device Attributes API.
+  Promise<DOMString> getAnnotatedAssetId();
+  Promise<DOMString> getAnnotatedLocation();
+  Promise<DOMString> getDirectoryId();
+  Promise<DOMString> getHostname();
+  Promise<DOMString> getSerialNumber();
+};
+</xmp>
+ 
+Methods on this interface typically complete asynchronously, queuing work on the <dfn>managed data task source</dfn>.
+ 
+## {{getAnnotatedAssetId()}} method ## {#getAnnotatedAssetId-method}
+ 
+  The <dfn method for="NavigatorManagedData">getAnnotatedAssetId()</dfn> method steps are:
+  1. Let |promise| be [=a new promise=].
+  1. Run the following steps in parallel:
+      1. If this API is not called by a managed web application with permission to access device attributes, reject |promise| with a {{NotAllowedError}} DOMException.
```

Comment:
"a managed web application with permission to access device attributes" ought to have a definition of some sort. Is the model that the device administrator maintains a list of origins that are allowed to access device attributes, and we check that the global object's origin is on that list?

Reply:
I agree. When moving some of the description of the model from the "Privacy considerations" section into the "Model" section we can explain this.

Reply:
I added a seperated 'Permission Control' paragraph in the Model section, and re-organize the description according to my personal understanding. Please help to check whether it looks better. Thanks!